cmake/CeleritasLibrary.cmake

```
@@ -83,6 +128,37 @@ define_property(TARGET PROPERTY CELERITAS_CUDA_OBJECT_LIBRARY
   FULL_DOCS "Name of the object (without nvlink step) library corresponding to this cuda library"
 )
 
+##############################################################################
+# Separate the OPTIONS out from the sources
+#
+macro(CUDA_GET_SOURCES_AND_OPTIONS _sources _cmake_options _options)
+  set( ${_sources} )
+  set( ${_cmake_options} )
+  set( ${_options} )
+  set( _found_options FALSE )
+  foreach(arg ${ARGN})
```

If you change this from a macro to a function you can do:

```suggestion
  foreach(arg IN LISTS ARGN)
```

I think it would be clealer to use a function and `PARENT_SCOPE` than having these scopeless underscore variables and the `set(` functions to clear them.

---

https://github.com/Kitware/CMake/blob/b8b53db1b30d697980bbf361a3f2604e1ff8396b/Modules/FindCUDA.cmake#L1327